services/kvstore/src/main/java/org/hyperledger/besu/services/kvstore/LayeredKeyValueStorage.java

```diff
@@ -178,61 +178,60 @@ public Stream<Pair<byte[], byte[]>> stream(final SegmentIdentifier segmentId) {
     throwIfClosed();
     var ourLayerState = hashValueStore.computeIfAbsent(segmentId, s -> newSegmentMap());
 
-    // otherwise, interleave the sorted streams:
-    final PeekingIterator<Map.Entry<Bytes, Optional<byte[]>>> ourIterator =
-        new PeekingIterator<>(
-            ourLayerState.entrySet().stream()
-                .filter(entry -> entry.getValue().isPresent())
```

Comment:
Removing this is the core change of this PR, right

This is a generic keyvalue layered storage facade, but LayeredKV storage exists essentially exclusively for bonsai.  I cannot think of a situation where we would want to filter out empty here and defer to the parent in the case of worldstate.